### PR TITLE
holiday booking is possible also tests

### PR DESCRIPTION
**Pull request by not-enough-ram**

### CODE REVIEW

File: backend/src/test/java/de/neuefische/backend/service/HolidaysServiceTest.java [hunk 1]
```diff
@@ -48,4 +59,210 @@ void getListOfHolidaysShouldReturnAListOfAllHolidaysInRepository() {
...
+                .startDate(LocalDate.of(2021, 10, 1))
+                .endDate(LocalDate.of(2022, 11, 17))
+                .build());
+
+        when(bookedHolidaysRepository.findById("testuser")).thenReturn(Optional.of(BookedHolidays.builder()
+                .user(User.builder()
+                        .appUser(AppUser.builder()
+                                .username("testuser")
+                                .password("")
+                                .build())
```
Comment (christophersiem) [inline backend/src/test/java/de/neuefische/backend/service/HolidaysServiceTest.java:131]: I think I would extract the AppUser builder

**Review by christophersiem — COMMENTED** [review]
Added a few comments. Try to make your pull request smaller :)

File: backend/src/main/java/de/neuefische/backend/service/HolidaysService.java [hunk 2]
```diff
@@ -37,4 +49,41 @@ public Holidays dtoToHolidaysObject(HolidaysDto holidaysDto){
...
+        return null;
+    }
+
+    public BookedHolidays setBookedHolidays(BookedHolidaysDto bookedHolidaysDto, String user) {
+        if (bookedHolidaysRepository.findById(user).isPresent()){
+            return bookedHolidaysRepository.save(dtoTobookedHolidays(bookedHolidaysDto, user));
+        }
+        else{
+            return bookedHolidaysRepository.save(dtoTobookedHolidays(bookedHolidaysDto, user));}
+    }
```
[the remark this answers is not in the document]
Comment (not-enough-ram) [reply backend/src/main/java/de/neuefische/backend/service/HolidaysService.java:65]: yeah... had no update function ready at point of pr but wanted to push anyway. should be better by now

File: backend/src/test/java/de/neuefische/backend/service/HolidaysServiceTest.java (hunk 1 above)
Comment (not-enough-ram) [reply backend/src/test/java/de/neuefische/backend/service/HolidaysServiceTest.java:131]: changed db model, now only login is needed

File: backend/src/main/java/de/neuefische/backend/dto/BookedHolidaysDto.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.NoArgsConstructor;
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class BookedHolidaysDto {
+
+    private String holidaysName;
+    private String startDateBooking;
+    private String endDateBooking;
```
[the remark this answers is not in the document]
Comment (not-enough-ram) [reply backend/src/main/java/de/neuefische/backend/dto/BookedHolidaysDto.java:14]: looked it up, not sure what it does

**Pull request merged**